Source/WebCore/page/Navigation.cpp

```diff
@@ -410,7 +410,6 @@ bool Navigation::innerDispatchNavigateEvent(NavigationNavigationType navigationT
     bool isTraversal = navigationType == NavigationNavigationType::Traverse;
     bool canIntercept = documentCanHaveURLRewritten(document, destination->url()) && (!isTraversal || isSameDocument);
     bool canBeCanceled = !isTraversal || (document->isTopDocument() && isSameDocument); // FIXME: and either userInvolvement is not "browser UI", or navigation's relevant global object has transient activation.
-    bool hashChange = equalIgnoringFragmentIdentifier(document->url(), destination->url()) && !equalRespectingNullity(document->url().fragmentIdentifier(),  destination->url().fragmentIdentifier());
```

Comment:
I'd prefer to keep this check here. We should always have access to the correct URLs.

Even if it means doing the same check twice here it probably won't whenever we have dispatches elsewhere for cross-document navigations.

Reply:
Done.